api-client/spec/api/extensions/get_into_teaching_api_client/api_client_spec.rb

```diff
@@ -171,4 +172,90 @@ def perform_post_request
 
     expect { perform_get_request }.to raise_error(Faraday::ConnectionFailed)
   end
+
+  describe "Circuit breaker" do
+    let(:threshold) { 3 }
+    let(:timeout) { 5.minutes }
+
+    context "when enabled" do
+
+      before(:each) do
+        Stoplight::Light.default_data_store = Stoplight::DataStore::Memory.new
+
+        GetIntoTeachingApiClient.configure do |config|
+          config.circuit_breaker = { enabled: true, threshold: threshold, timeout: timeout }
+        end
+      end
+
+      context "when the API returns a single error that can cause a broken circuit" do
+        it "does not break the circuit" do
+          stub_request(:get, get_endpoint)
+            .to_return(status: 500)
+
+          expect { perform_get_request }.to raise_error(GetIntoTeachingApiClient::ApiError)
+        end
+      end
+
+      context "when the API returns a number of errors above the threshold that can cause a broken circuit" do
+        it "breaks the circuit" do
+          stub_request(:get, get_endpoint)
+            .to_return(status: 500)
+
+          threshold.times do
+            expect { perform_get_request }.to raise_error { |error|
```

Comment:
I think this (and other instances) can be simplified to match line 195 now as the code is identical? so just:

```
expect { perform_get_request }.to raise_error(GetIntoTeachingApiClient::ApiError)
```